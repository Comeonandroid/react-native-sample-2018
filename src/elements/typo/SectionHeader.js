import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { Text } from '../index'


const SectionHeader = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

SectionHeader.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  margin: 5px 0px;
  font-family: Lato;
  font-weight: 400;
  font-size: 15px;
`

export default SectionHeader