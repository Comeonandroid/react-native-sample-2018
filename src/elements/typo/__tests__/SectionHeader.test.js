import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import SectionHeader from '../SectionHeader'

describe('<SectionHeader />', () => {
  itBehavesLike('aThemedComponent', { subject: <SectionHeader /> })
  itBehavesLike('aComponentWithChildren', { subject: <SectionHeader /> })
})
