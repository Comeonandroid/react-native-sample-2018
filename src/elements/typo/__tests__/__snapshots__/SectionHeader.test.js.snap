// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionHeader /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fcfcfc",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "fontFamily": "Lato",
          "fontSize": 15,
          "fontWeight": "400",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<SectionHeader /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "fontFamily": "Lato",
          "fontSize": 15,
          "fontWeight": "400",
          "marginBottom": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 5,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<SectionHeader /> renders component when passed in 1`] = `
<Styled(Text)>
  <Component />
</Styled(Text)>
`;

exports[`<SectionHeader /> renders text when passed in 1`] = `
<Styled(Text)>
  Test
</Styled(Text)>
`;
